packages/desktop-client/src/components/reports/ReportOptions.tsx

```diff
@@ -44,3 +51,132 @@ const intervalOptions = [
 { value: 5, description: 'Yearly', name: 5,
 ];
 */
+export type QueryDataEntity = {
+  date: string;
+  category: string;
+  categoryGroup: string;
+  account: string;
+  accountOffBudget: boolean;
+  payee: string;
+  transferAccount: string;
+  amount: number;
+};
+
+export type UncategorizedEntity = {
+  id?: string;
+  name: string;
+  hidden?: boolean;
+  uncat_id: string;
+  offBudget: boolean;
+  transfer: boolean;
+  category: boolean;
+};
+
+export type UncategorizedGroupEntity = {
+  id?: string;
+  name: string;
+  hidden?: boolean;
+  categories?: UncategorizedEntity[];
+};
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
```suggestion
export type UncategorizedGroupEntity = CategoryGroupEntity & {
  id: null;
  categories?: UncategorizedEntity[];
};
```